x-pack/filebeat/input/azureblobstorage/config.go

```diff
@@ -6,15 +6,32 @@ package azureblobstorage
 
 import (
 	"errors"
-	"reflect"
 	"time"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 
 	"github.com/elastic/beats/v7/libbeat/common/match"
 	"github.com/elastic/beats/v7/libbeat/reader/parser"
+	conf "github.com/elastic/elastic-agent-libs/config"
 )
 
+// defaultReaderConfig is a default readerConfig state that is used to evaluate
+// if the container level ReaderConfig is explicitly configured by the user or not.
+// If it is not configured, the container level ReaderConfig is ignored and the global
+// ReaderConfig is used instead.
```

Comment:
```suggestion
// defaultReaderConfig is a default readerConfig state that is used to evaluate
// if the container level ReaderConfig is explicitly configured by the user or not.
// It must not be mutated.
```
(the place for the last sentence in the original is in the site of use)